### DISCUSSION TITLE
Using pyodide to build an online code editor

### DISCUSSION THREAD

**Discussion by ChristopherHButler** [pyodide/pyodide#1242] [Q&A] [answered]
Hello, 
I am currently building an online code editor (something like code sandbox) and I would like to support python. After reading [this article](https://yasoob.me/2019/05/22/running-python-in-the-browser/) pyodide seems the like best option for my use case but I was looking for someone to validate that assumption before I start down a potentially long path 😄
For reference, this is my project in its current state: [https://contrived.herokuapp.com/](https://contrived.herokuapp.com/)
Any feedback is greatly appreciated :)
Thanks!

**Comment by rth** [comment, marked as answer]
> pyodide seems the like best option for my use case but I was looking for someone to validate that assumption

As a pyodide maintainer I can give you my fully impartial opinion that it is indeed the best solution :)

More seriously though, there are now various web IDE related solutions built on top of pyodide (for instance https://notebook.basthon.fr/ , https://starboard.gg/#python and [iodide](https://github.com/iodide-project/iodide)), so it seems to work reasonably well for that use case. Maybe @casatir could share his experience with trying alternative Python solutions for Basthon.

**Reply by ChristopherHButler** [reply]
Awesome. Thanks Roman!

**Reply by casatir** [reply]
Thanks @rth for linking to Basthon.

You will find useful info about it [here](https://basthon.fr). Basthon has also a [console based IDE](https://console.basthon.fr).

I made a comparison [here](https://basthon.fr/about.html#others) (in french). While being really great projects, Brython and Skulpt were not as powerful as Pyodide at the time I've tested them (~1 year ago). Basthon has been written first using Brython then using Skulpt and finally with Pyodide so I'm pretty aware.

I can comment deeper on this if needed.

**Reply by ChristopherHButler** [reply]
Thanks @casatir I appreciate the info. I've had a quick look at the [docs](https://pyodide.readthedocs.io/en/latest/usage/quickstart.html) and as I mentioned in the OP I believe pyodide will suit my needs. I may need to adjust some things for my editor to make a bit more sense running python however. My editor currently support several js frameworks meant to create front end projects where the app's "browser" updates in real time with every key press. I also have a javascript repl of sorts where you can write some javascript code and just evaluate it and I imagine this is how I will support python. What would be interesting though, is that since I can evaluate python code and get the result, I could use the results directly in a js framework like react to present the findings of a data science project in real time in a browser. 🤯
